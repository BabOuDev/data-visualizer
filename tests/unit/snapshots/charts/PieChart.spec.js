import {shallowMount} from '@vue/test-utils';
import PieChart from '@/components/charts/PieChart.vue';

describe('PieChart.vue', () => {
  it('should renders correctly', () => {
    const component = shallowMount(PieChart, {
      props: {
        rows: [
          {colName: 'A'},
          {colName: 'B'},
          {colName: 'B'},
          {colName: 'C'},
          {colName: 'C'},
          {colName: 'C'},
        ],
        col: {path: 'colName'},
      },
    });
    expect(component.html()).toMatch('A');
    expect(component.html()).toMatch('B');
    expect(component.html()).toMatch('C');
    expect(component.html()).toMatchSnapshot();
  });
});
