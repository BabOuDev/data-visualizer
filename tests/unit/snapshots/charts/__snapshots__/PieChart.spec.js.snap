// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart.vue should renders correctly 1`] = `
<div class="pie-container">
  <legend-stub keys="A,B,C" colors="#42b883,#3c8171,#35495e"></legend-stub>
  <div class="pie-background"></div>
  <div style="transform: rotate(0deg);" class="hold" title="A: 1">
    <div class="pie" style="background-color: rgb(66, 184, 131); transform: rotate(60deg);"></div>
  </div>
  <!--v-if-->
  <div style="transform: rotate(60deg);" class="hold" title="B: 2">
    <div class="pie" style="background-color: rgb(60, 129, 113); transform: rotate(120deg);"></div>
  </div>
  <!--v-if-->
  <div style="transform: rotate(180deg);" class="hold" title="C: 3">
    <div class="pie" style="background-color: rgb(53, 73, 94); transform: rotate(180deg);"></div>
  </div>
  <!--v-if-->
  <div class="inner-circle">
    <div class="content"><b></b></div>
  </div>
</div>
`;
